src/languages/latex.ts

```diff
@@ -0,0 +1,171 @@
+  "color_set_definition",
+  "color_reference",
+  "tikz_library_import",
+];
+
+const GROUPS = [
+  "curly_group",
+  "curly_group_text",
+  "curly_group_text_list",
+  "curly_group_path",
+  "curly_group_path_list",
+  "curly_group_command_name",
+  "curly_group_key_value",
+  "curly_group_glob_pattern",
+  "curly_group_impl",
+  "brack_group",
+  "brack_group_text",
+  "brack_group_argc",
+  "brack_group_key_value",
+  "mixed_group",
+];
+
+const SECTIONING = [
+  "part",
+  "chapter",
+  "section",
+  "subsection",
+  "subsubsection",
+  "paragraph",
+  "subparagraph",
+];
+
+const ENVIRONMENTS = [
+  "generic_environment",
+  "comment_environment",
+  "verbatim_environment",
+  "listing_environment",
+  "minted_environment",
+  "pycode_environment",
+];
+
+const sectioningText = SECTIONING.map((s) => `${s}[text]`);
+const sectioningCommand = SECTIONING.map((s) => `${s}[command]`);
+const commandXGroup = GROUPS.map((group) =>
+  COMMANDS.map((cmd) => `${cmd}.${group}!`)
+).flat();
+
+function unwrapParenExtractor(
+  editor: TextEditor,
+  node: SyntaxNode
+): SelectionWithContext {
+  let startIndex = node.startIndex;
+  let endIndex = node.endIndex;
+  const parens = [
+    ["{", "}"],
+    ["[", "]"],
+  ];
+  if (
+    parens.some(
+      ([left, right]) => node.text.startsWith(left) && node.text.endsWith(right)
+    )
+  ) {
+    startIndex += 1;
+    endIndex -= 1;
+  }
+  return {
+    selection: new Selection(
+      editor.document.positionAt(startIndex),
+      editor.document.positionAt(endIndex)
+    ),
+    context: {},
+  };
+}
+
+function extendToNamedSiblingIfExists(
+  editor: TextEditor,
+  node: SyntaxNode
+): SelectionWithContext {
+  let startIndex = node.startIndex;
+  let endIndex = node.endIndex;
+  let sibling = node.nextNamedSibling;
+
+  if (sibling != null && sibling.isNamed) {
+    endIndex = sibling.endIndex;
+  }
+
+  return {
+    selection: new Selection(
+      editor.document.positionAt(startIndex),
+      editor.document.positionAt(endIndex)
+    ),
+    context: {},
+  };
+}
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  argumentOrParameter: matcher(
+    patternFinder(
+      ...commandXGroup
+        .concat(["begin[name]", "end[name]"])
+        .concat(sectioningText)
+    ),
+    unwrapParenExtractor
+  ),
+
+  functionCall: cascadingMatcher(
+    matcher(patternFinder(...COMMANDS.concat(["begin", "end"]))),
+    matcher(patternFinder(...sectioningCommand), extendToNamedSiblingIfExists)
+  ),
+
+  name: matcher(patternFinder(...sectioningText), unwrapParenExtractor),
+  type: "command_name",
+
+  collectionItem: "enum_item",
+
+  comment: ["block_comment", "line_comment"],
+
+  // TODO: use other scope(s)? adding individual scopes is probably too much? (part, chapter, section, subsection, subsubsection, paragraph, subparagraph)
+  xmlStartTag: SECTIONING,
+
+  // TODO: eventually add scope instead of reusing xmlElement
+  xmlElement: ENVIRONMENTS,
```

Comment:
I am happy for you to be liberal in adding scopes.  This language doesn't really fit into the paradigm of the existing implemented languages.  We'll soon be moving to a more flexible scope model.  So for now just add as many as you want.  If it gets out of hand we can trim it down